automerge-backend/src/change.rs

```diff
@@ -171,6 +173,54 @@ fn encode_chunk(change: &amp::Change, deps: &[amp::ChangeHash]) -> ChunkIntermed
     }
 }
 
+/// When encoding a change we take all the actor IDs referenced by a change and place them in an
+/// array. The array has the actor who authored the change as the first element and all remaining
+/// actors (i.e. those referenced in object IDs in the target of an operation or in the `pred` of
+/// an operation) lexicographically ordered following the change author.
+///
+/// Lexicographic order here means the lexigographic ordering of the lower case hex encoding of the
+/// actor IDs.
+fn actor_ids_in_change(change: &amp::Change) -> Vec<amp::ActorId> {
+    // We construct a vector of HexActorIds to make use of it's Ord implementation which compares
+    // actor IDs lexicographically
+    let mut other_ids: Vec<HexActorId<'_>> = change
+        .operations
+        .iter()
+        .flat_map(opids_in_operation)
+        .filter(|a| *a != &change.actor_id)
+        .map(|a| a.into())
+        .unique()
+        .collect();
+    other_ids.sort();
+    // Now prepend the change actor and convert all the HexActorIds back into ActorIds
+    Some(&change.actor_id)
```

Comment:
Could this use https://doc.rust-lang.org/std/iter/fn.once.html instead of the option?